biodatasets/tmvar_v3/tmvar_v3.py

```diff
@@ -0,0 +1,306 @@
+                            "offsets": datasets.Sequence([datasets.Value("int32")]),
+                            "semantic_type_id": datasets.Sequence(
+                                datasets.Value("string")
+                            ),
+                            "normalized": {
+                                key: datasets.Sequence(datasets.Value("string"))
+                                for key in type_to_db_mapping.keys()
+                            },
+                        }
+                    ],
+                }
+            )
+        elif self.config.schema == "bigbio_kb":
+            features = schemas.kb_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        url = _URLS[_DATASETNAME]
+        test_filepath = dl_manager.download(url)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": test_filepath,
+                },
+            )
+        ]
+
+    def get_normalizations(self, id, type):
+        """
+        Given a type and a number of normalizations ids, this function returns a dictionary of the normalized ids
+        """
+        base_dict = {
+            key: []
+            for key in [
+                "tmVar",
+                "CorrespondingGene",
+                "dbSNP",
+                "VariantGroup",
+                "NCBI Taxonomy",
+            ]
+        }
+        ids = id.split(";")
+        if type in ["CellLine", "Species"]:
+            id_vals = ids[0].split(",")
+            base_dict["NCBI Taxonomy"] = id_vals
+
+        elif type == "Gene":
+            id_vals = ids[0].split(",")
+            base_dict["CorrespondingGene"] = id_vals
+
+        else:
+            for id in ids:
+                if "|" in id:
+                    base_dict["tmVar"].append(id)
+                elif id[:2] == "rs":
+                    base_dict["dbSNP"].append(id[2:])
+                elif ":" in id:
+                    db_name, db_id = id.split(":")
+                    if db_name == "RS#":
+                        db_name = "dbSNP"
+                    # Hacky fix below for doc ID: 18272172
+                    elif db_name == "Va1iantGroup":
+                        db_name = "VariantGroup"
+                    elif db_name == "Gene":
+                        db_name = "CorrespondingGene"
+                    elif db_name == "Disease":
+                        continue
+                    base_dict[db_name].append(db_id)
+                else:
+                    logger.warn(f"Malformed normalization. Type: {type}, Number: {id}")
```

Comment:
Would it be possible to add SPLIT and DOCUMENT_ID to the warning? This way it's easier to pinpoint them

Reply:
This dataset only has the Test SPLIT, but I can easily add the document ID. Should I add test split as a deafult or leave it out?

Reply:
I have added document ID to the warning

Reply:
> This dataset only has the Test SPLIT

Pfff, you are right! Thanks!